packages/next/src/server/next-server.ts

```diff
@@ -1605,7 +1605,12 @@ export default class NextNodeServer extends BaseServer<
 
       result = await adapterFn({
         handler: middlewareModule.middleware || middlewareModule,
-        request: { ...requestData, body: requestData.body?.cloneBodyStream() },
+        request: {
+          ...requestData,
+          body: !['HEAD', 'GET'].includes(params.request.method)
+            ? requestData.body?.cloneBodyStream()
+            : undefined,
```

Comment:
This is already handled [here](https://github.com/kevva/next.js/blob/89fc83b0c3a601e8c6c5d07d42cd674439dc8ed6/packages/next/src/server/next-server.ts#L1579-L1582) so this check is kinda unnecessary.

Reply:
Partially yeah just for consistency as the sandbox does the same check again as well.

Reply:
Right, with the small difference that sandbox is the one doing the cloning whereas here we pass it cloned already. But makes sense if you want to move this part in the future.

Reply:
@ijjk, actually, this is what causes https://github.com/vercel/next.js/issues/77646 and https://github.com/vercel/next.js/pull/77553#issuecomment-2765993539.